### DISCUSSION TITLE
Anzeige von Thermostaten

### DISCUSSION THREAD

**Discussion by eikowagenknecht** [SukramJ/aiohomematic#314] [Ideas]
Hallo zusammen,

in einer der letzten Versionen hat sich anscheinend etwas in der Darstellung der Heizthermostate geändert. Da es nur für die Homematic Heizungen gilt, nehme ich mal an, dass ihr das wart ;-)

So sieht es nun bei mir aus:
![image](https://user-images.githubusercontent.com/1475672/155843117-14689243-d5ca-456f-99f8-3066925af2ca.png)

Rot markiert sind die Homematic Geräte.

Ich finde es ja grundsätzlich begrüßenswert, viele Informationen darzustellen, aber hier habe ich das Gefühl, dass es etwas "too much" ist. **Auf meinem iPhone kann ich so z.B. beim Abstellraum (oben rechts im Bild) nur noch die ersten 2 Buchstaben des Raumes lesen**, bevor der Status losgeht.

Mir scheint, dass die Logik aktuell wie folgt ist:
- Position 1: Aktueller Status
  - "Heizung" bei Ventilöffnung >0
  - "Leerlauf" bei Ventilöffnung 0
- Position 2: Eingestellter Modus
  - "(Heizen)" bei manuellem Betrieb
  - "(Automatisch)" bei Automatik
- Position 3: Aktiviertes Profil (nur bei Automatik)
  - "- Profile 1" bei Profil 1, etc.

Daran stört mich:
- Die Benennung ist nicht einheitlich (Heizung / Heizen)
- Die Informationen sind teils redundant
- Profile ist nicht auf deutsch übersetzt im Gegensatz zu den anderen Einträgen

Ich könnte es mir besser so vorstellen:
- "Heizen" bei Ventilöffnung >0 und Modus manuell
- "Automatisch" bei Ventilöffnung >0 und Modus Automatik (ggf. auch "Automatisch - Profil 1")
- "Leerlauf" bei Ventilöffnung 0

Oder noch einfacher:
- "Heizen" bei Modus manuell
- "Automatisch" bei Modus Automatik (ggf. auch "Automatisch - Profil 1")

Wer die Ventilöffnung spannend findet, kann sie ja einfach einblenden, wie ich das z.B. in meinem Screenshot auch mache.

Ein kleiner Bug ist mir dabei auch noch aufgefallen, siehe #315.

Was denkt ihr dazu?

**Comment by Lemocin** [comment]
> Die Benennung ist nicht einheitlich (Heizung / Heizen)

Ich weiß nicht, in wiefern Homeassistant, das "Heizung" automatisch anzeigt, zumindest benutzt es das intern für den Verlauf, was ich ganz nett finde:
![grafik](https://user-images.githubusercontent.com/4162496/155844750-ddf85114-3f63-4ad3-a0e5-751f2fd30f71.png)

Homeassistant schneidet den Namen bei mir auch ab, wobei ich allerdings finde, dass das im Frontend selbst behoben werden sollte. Gibt ja auch andere Integrationen mit längeren Statusanzeigen... Auf meinem Smartphone mit 6,6" Display wird mir z.B. nur ein Buchstabe angezeigt. Dafür habe ich mir aber so eine Climate Karte erstellt, die allerdings auch viel Platz weg nimmt.

> Die Informationen sind teils redundant

Redundant ist mir hier allerdings nichts aufgefallen. 
Leerlauf -> Freie stelle im Verlauf, Thermostat heizt nicht. Heizung -> gelbe Markierung im Verlauf, Thermostat heizt.
Heizen -> manueller Modus. Aus -> ausgeschaltet. Automatisch -> Zeitprogramm
Und dann noch die Profilanzeige von Profile 1 bis zu Boost


> Profile ist nicht auf deutsch übersetzt im Gegensatz zu den anderen Einträgen

Kann ich bestätigen


> "Heizen" bei Ventilöffnung >0 und Modus manuell

Statt Heizen könnte man auch manuell schreiben, das fände ich etwas besser. Dann würde es eine klare Abgrenzung zwischen dem manuellen und dem Automatischen Modus geben. 


> * "Automatisch" bei Ventilöffnung >0 und Modus Automatik (ggf. auch "Automatisch - Profil 1")
> 
> * "Leerlauf" bei Ventilöffnung 0

Im Zeitprogramm würde ich sagen, wäre es besser, es bei Automatisch zu belassen. Sonst ist ja unklar, ob man jetzt im manuellen oder automatischen Modus ist etc.. 

Haben deine gelb markierten Thermostate auch Zeitprogramme, oder einen Status ob sie gerade Heizen im Verlauf?

**Reply by eikowagenknecht** [reply]
Der gelbe Eintrag kommt von Melcloud, ist also eine Klimaanlage und keine Heizung. Der Verlauf sieht wie folgt aus: 
![image](https://user-images.githubusercontent.com/1475672/155848530-7e361016-0f20-4aab-a616-0ca690f6bd80.png)
Gelbe Flächen sind dabei "Heizen", dazwischen "Aus".

**Comment by SukramJ** [comment]
Es gibt bei HA den Heizmodus (hvac_mode) (u.a. Automatisch, Heizen und Aus), und es gibt die aktuelle Aktion (hvac_action) z.Bsp. Ventil geöffnet zeigt Heizen an ansonsten Leerlauf.
Also die Anzeige besteht aus `hvac_action(hvac_mode) - preset_mode`.

Im Code werden nur Konstanten (hvac_mode,hvac_action, preset_mode) verwendet, d.h. die Übersetzungen steuert das Frontend bei.
Der preset_mode enthält HM spezifische Werte (Profile 1 - Profile 6) wird nur als Text übergeben und nicht übersetzt, so weit ich weiss gibt es hier auch keine Übersetzungsmöglichkeiten. 

Also wenn Euch Übersetzungen nicht gefallen oder Texte zu lang oder abgeschnitten sind, dann müsst ihr ein Ticket im HA-Frontend Repo aufmachen, ggf. könnte man Profile 1 - Profile 6 auch einfach P1-P6 benennen.

**Reply by eikowagenknecht** [reply]
Ah, das heißt die Darstellung der Werte (`hvac_action(hvac_mode) - preset_mode`)kommt komplett von HA und nicht von der Integration? Dann sind meine Anmerkungen hier natürlich fehl am Platz. Ich war davon ausgegangen, dass das in der Integration gehandhabt wird weil meine anderen Heizungen / Klimaanlagen deutlich weniger Text anzeigen an der Stelle.

**Comment by Aearon** [comment]
Interessant - ich würde allerdings gerne die Ventilstellung sehen bzw auch als separaten Sensor haben. Wie bekomme ich das denn in der aktuellen Version hin?

@eikowagenknecht War das bei der by default so?

**Reply by eikowagenknecht** [reply]
Ich nutze dafür folgendes in einer normalen Liste als Element: https://github.com/custom-cards/secondaryinfo-entity-row

```
entity: climate.eg_bad_heizung
name: Bad
type: custom:secondaryinfo-entity-row
secondary_info: >-
  {% if states("number.eg_bad_heizung_ventiloeffnung") and
  (states("number.eg_bad_heizung_ventiloeffnung") | float) %}
    {{(states("number.eg_bad_heizung_ventiloeffnung") | float) | int}}
  %{%endif%}
```

Die Ventilöffnungs Entität ist im Standard dabei:
![image](https://user-images.githubusercontent.com/1475672/165073381-9461b9f0-53b4-4f69-b8bc-6fd41aa7d26b.png)

**Reply by Aearon** [reply]
Besten dank für die schnelle Antwort. Das entscheidende war, dass die _level Entitäten standardmäßig versteckt waren. Auf die war ich bisher noch nicht gestoßen.

**Reply by emufan** [reply]
> secondary_info: >-
>   {% if states("number.eg_bad_heizung_ventiloeffnung") and
>   (states("number.eg_bad_heizung_ventiloeffnung") | float) %}
>     {{(states("number.eg_bad_heizung_ventiloeffnung") | float) | int}}
>   %{%endif%}

Warum machst Du das so? Was soll if die and-Abfrage und was das |float|int? Oder war das mal anders und sind nur Überbleibsel aus alten Varianten, die momentan eigentlich keinen aktuell Sinn ergeben?

**Reply by eikowagenknecht** [reply]
Ich hätte die Anzeige gerne ohne Nachkokmmastelle. Manchmal ist in der Startphase von HA das Gerät nicht verfügbar und HA meckert dann, dass man None nicht in int umwandeln kann und weigert sich komplett, das Template darzustellen. Mit dem Code von mir funktioniert es. Hast du eine einfachere Schreibweise dafür? Nehme ich gerne :-)

**Reply by emufan** [reply]
Dafür ist das default beim int da und bald sogar mandatory. |int(default=0). Aber nichts anzuzeigen, wenn nicht vorhanden, finde ich auch nicht schlecht. Dennoch benötigt es mE kein and und keine Doppelkonvertierung zu Kommawert und dann zu Ganzzahl.

```
{% if states("number.eg_bad_heizung_ventiloeffnung") %}
  {{ states("number.eg_bad_heizung_ventiloeffnung") | int(default=0) }} %
{% endif %}
```

**Reply by eikowagenknecht** [reply]
Nope, das zeigt "0 %" bei Thermostaten die aus sind. 

Vielleicht geht das hier?

```
{% if states("number.eg_bad_heizung_ventiloefffnung") | int(default=0) %}
{{ states("number.eg_bad_heizung_ventiloeffnung") | int(default=0) }} %
{%endif%}
```

**Reply by emufan** [reply]
Kommt halt auf den Inhalt an, wenn aus ist. Guck mal in Developer Tools und prüfe dagegen, z.B. bei unavailable wenn aus

```
{% if states("number.eg_bad_heizung_ventiloeffnung") != "unavailable" %}
  {{ states("number.eg_bad_heizung_ventiloeffnung") | int(default=0) }} %
{% endif %}
```
